### ISSUE TITLE
Available money on 'CSEC' grown by NaN%.

### ISSUE THREAD

**Issue by bartekleon** [bitburner-official/bitburner-src#864]
- [bitburnerSave_1696773466_BN1x1.json.txt](https://github.com/bitburner-official/bitburner-src/files/12840863/bitburnerSave_1696773466_BN1x1.json.txt) Save file 
- Minimal scripts to reproduce the issue [miner.js.txt](https://github.com/bitburner-official/bitburner-src/files/12840866/miner.js.txt)
[spreader.txt](https://github.com/bitburner-official/bitburner-src/files/12840867/spreader.txt)
- Steps to reproduce
Running `spreader.js` on servers
- [2.5.0] Version of the game, e.g. Bitburner v1.3.0 (216bf616)

I noticed after some time that my miners don't produce money on some servers. It appears the servers can't grow.

Running manually `grow` on CSEC showed `Available money on 'CSEC' grown by NaN%.`

**Comment by Caldwell-74** [comment]
some servers have 0 max Money so you cant grow them

**Comment by d0sboots** [comment]
When I run your `miner.js` on CSEC, this is what I see:
```
grow: Executing on 'CSEC' in 6.073 seconds (t=1).
grow: Available money on 'CSEC' grown by -100.000000%. Gained 3.925 hacking exp (t=1).
hack: Executing on 'CSEC' in 1.897 seconds (t=1)
hack: Successfully hacked 'CSEC' for $0.000 and 0.981 exp (t=1)
hack: Executing on 'CSEC' in 1.898 seconds (t=1)
hack: Successfully hacked 'CSEC' for $0.000 and 0.981 exp (t=1)
hack: Executing on 'CSEC' in 1.899 seconds (t=1)
hack: Successfully hacked 'CSEC' for $0.000 and 0.981 exp (t=1)
hack: Executing on 'CSEC' in 1.900 seconds (t=1)
hack: Successfully hacked 'CSEC' for $0.000 and 0.981 exp (t=1)
hack: Executing on 'CSEC' in 1.901 seconds (t=1)
hack: Successfully hacked 'CSEC' for $0.000 and 0.981 exp (t=1)
```

This is also what I saw when I loaded your savegame. However, I did see this in your logs:
```
Called on CSEC 1194 times while offline
'CSEC' grown by NaN% while offline
Called weaken() on CSEC 145 times while offline
```

I'm guessing you mistyped the exact line when you made the bug report.

Regardless, as Caldwell says this isn't really a bug. CSEC (and other servers) are unable to be grown because they have 0 max money, so the most correct grow percent *is* NaN (0/0). -100% is actually kind of more wrong, but you're trying to grow (and hack) a server with no money, so the exact message isn't super critical.

**Comment by bartekleon** [comment]
Maybe for displaying it
isFinite(num) ? 0 : num
Or at least isNaN check

I do understand it is not a bug, but visually it doesn't look appealing.
Still better than getting random infinity
